### DISCUSSION TITLE
Direct Connection

### DISCUSSION THREAD

**Discussion by virusuf86** [jasonacox/Powerwall-Dashboard#349] [Ideas]
Is there a way to configure a RP4 wifi to direct connect to the inverter (SSID: TEG-xxx) and pull data for InfluxDB? The RP4 will also be plugged in via ethernet to access the internet/SSH/Dashboard.

I started with my RP4 on ethernet. My Powerwall and RP4 have a static IP. Sometimes when I reboot my RP4, it cant reestablish connection to the Powerwall. The Power Flow panel in the Dashboard does not load. I also noticed that the Tesla Energy - Setup page wont load via the Powerwall IP. To get it working again, I have to reboot the AP the ethernet is plugged into. The AP is running an older firmware since I have issues with the newer ones. At the end of the day, it could be the AP's fault. I switched my RP4 to wifi and so far no issues after a few reboots.

**Comment by jasonacox** [comment]
Yes, you should be able to do that. You would specify the address of the Powerwall as 192.168.91.1 (per [Tesla](https://www.tesla.com/support/energy/powerwall/own/connecting-network)) when doing the Dashboard setup.  As to how to set that up, there are several Raspberry Pi help pages out there, or you can ask ChatGPT (see [here](https://chat.openai.com/share/4197e583-a991-4c9e-9dc5-ac82ea1a6a9d)).

However, it has been my experience that the WiFi on the Powerwall gateway is not great. It sounds like your WiFi AP isn't great either. I would highly recommend that you hardwire the Powerwall gateway as well as your Raspberry Pi. Before I did that, even when it would connect and show strong signal I would see a lot of data dropout. After hardwiring, no issue.  It took a bit of work for me to run an ethernet cable to the odd spot in my garage where my Powerwall system was located, but it was worth it.

**Reply by virusuf86** [reply]
Ill stick to my RP4 on wifi for now. Good to know theres an option for both interfaces. Thanks!
